src/libsyntax/parse/parser.rs

```diff
@@ -1866,10 +1866,18 @@ impl<'a> Parser<'a> {
                            style: PathStyle,
                            enable_warning: bool)
                            -> PResult<'a, ()> {
+        use self::token::{BinOp, OpenDelim, Star, Brace};
+
         loop {
             segments.push(self.parse_path_segment(style, enable_warning)?);
 
-            if self.is_import_coupler() || !self.eat(&token::ModSep) {
```

Comment:
Could you keep `is_import_coupler` alive and used here (even if it's used only here).
It will likely be useful again for fixing https://github.com/rust-lang/rust/pull/45846/files#r150397110.